sql/src/main/java/io/crate/execution/dsl/projection/ColumnIndexWriterProjection.java

```diff
@@ -115,9 +124,41 @@ public ColumnIndexWriterProjection(RelationName relationName,
             for (int i = 0; i < mapSize; i++) {
                 allTargetColumns.add(Reference.fromStream(in));
             }
+
+            int outputSize = in.readVInt();
```

Comment:
Nodes running 4.1 or earlier won't send this, so we need to add some BWC handling here.

Reply:
Good catch, thank you1

Reply:
Actually the whole block is already covered in a ` if (in.getVersion().onOrAfter(Version.V_4_2_0))` statement.  Maybe this was not obvious from the change log. https://github.com/crate/crate/blob/720b84a7c13191c2fddac716de4e7e49ed70f603/sql/src/main/java/io/crate/execution/dsl/projection/ColumnIndexWriterProjection.java#L121

Reply:
Ah right, didn't see this in the diff.